Framework/API/test/WorkspaceListPropertyTest.h

```diff
@@ -0,0 +1,304 @@
+#ifndef MANTID_API_WORKSPACELISTPROPERTYTEST_H_
+#define MANTID_API_WORKSPACELISTPROPERTYTEST_H_
+
+#include <cxxtest/TestSuite.h>
+
+#include "MantidAPI/WorkspaceListProperty.h"
+#include "MantidAPI/Workspace.h"
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidAPI/IMDHistoWorkspace.h"
+#include "MantidAPI/IMDEventWorkspace.h"
+#include "MantidAPI/ITableWorkspace.h"
+#include "MantidAPI/IEventWorkspace.h"
+#include "MantidAPI/Algorithm.h"
+#include "MantidTestHelpers/FakeObjects.h"
+#include "MantidKernel/IValidator.h"
+#include "MantidKernel/DataItem.h"
+#include <vector>
+#include <gmock/gmock.h>
+
+using namespace Mantid::API;
+using namespace Mantid::Kernel;
+using Mantid::Kernel::DataItem;
+
+class WorkspaceListPropertyTest : public CxxTest::TestSuite {
+private:
+  /**
+  * Helper class used for Mocking validators.
+  */
+  class MockValidator : public Mantid::Kernel::IValidator {
+  public:
+    MockValidator() {}
+    virtual ~MockValidator() {}
+    MOCK_CONST_METHOD0(clone, IValidator_sptr());
+    MOCK_CONST_METHOD1(check, std::string(const boost::any &));
+  };
+
+  /**
+  * Helper class. Algorithms are instances of IPropertyManager
+  */
+  class MyAlgorithm : public Mantid::API::Algorithm {
+  public:
+    MyAlgorithm() { this->setRethrows(true); }
+
+    virtual int version() const { return 1; }
+
+    virtual const std::string name() const { return "MyAlgorithm"; }
+
+	const std::string summary() const override { return "MyAlgorithm helper."; }
+
+    virtual void init() {
+      declareProperty(std::make_unique<WorkspaceListProperty<Workspace>>(
+          "MyProperty", std::vector<Workspace_sptr>(0)));
+    }
+
+    virtual void exec() {
+      std::vector<boost::shared_ptr<DataItem>> val = getProperty("MyProperty");
+    }
+
+    virtual ~MyAlgorithm() {}
+  };
+
+public:
+  // This pair of boilerplate methods prevent the suite being created statically
+  // This means the constructor isn't called when running other tests
+  static WorkspaceListPropertyTest *createSuite() {
+    return new WorkspaceListPropertyTest();
+  }
+  static void destroySuite(WorkspaceListPropertyTest *suite) { delete suite; }
+
+  //------------------------------------------------------------------------------
+  // Functional Testing
+  //------------------------------------------------------------------------------
+
+  void test_construction_minimalistic() {
+    AnalysisDataServiceImpl &ads = AnalysisDataService::Instance();
```

Comment:
I think we suggested that the ADS stuff was going to go (you're still working on this I think) in which case the constructor taking a workspace name must also be removed.